### PR TITLE
Plugin for concatenating mappings and generating a monitoring nomad job

### PR DESCRIPTION
**Pull request by Talina06**
Basic version, where I don't check if the mappings are in a valid YAML format. It also doesn't check the type of file, if it is binary that is being read or a plain text file. Suggest changes as you feel.
2nd: the command/plugin name can change, suggest a better one.

Usage:
```
$: escher monitor --directory=~/mappings --template-url=<path-to-base-monitoring-template>
Output: monitoring job saved in monitoring.nomad.j2
```

### CODE REVIEW

File: plugins/monitor/tmpl/monitoring.nomad.tmpl [hunk 1]
```diff
@@ -0,0 +1,416 @@
+job "monitoring" {
```
Comment (mohandutt134) [inline plugins/monitor/tmpl/monitoring.nomad.tmpl]: why is this file needed in repo if its fetched from the artifactory?

File: plugins/monitor/monitor.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		log.Printf("monitoring job saved in %v", OUTPUT)
+	default:
+		panic("unsupported command: " + conf.Command)
+	}
+
+	return nil
+}
+
+type Mappings struct {
+	New_mappings []string
```
Comment (mohandutt134) [inline plugins/monitor/monitor.go]:
```suggestion
	Mappings []string
```

File: plugins/monitor/monitor.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		// display that the job is saved to a file.
+		log.Printf("monitoring job saved in %v", OUTPUT)
+	default:
+		panic("unsupported command: " + conf.Command)
+	}
+
+	return nil
+}
+
+type Mappings struct {
```
Comment (mohandutt134) [inline plugins/monitor/monitor.go]:
```suggestion
type MappingData struct {
```

File: plugins/monitor/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Save all mapping YAML files for statsd in a directory on your local machine. Ex: folder name is ~/mappings.
+
+Assuming the monitor.so plugin is saved in your machine at path ~/.escher/plugins and the latest escher binary has been downloaded from github.com/tsocial/escher/releases,
+
+
+### Usage:
+
+```
+$: escher monitor --directory=~/mappings --template-url=<s3-tmpl-url>
```
Comment (mohandutt134) [inline plugins/monitor/README.md]: should take only template version, complete url will be formed using `artifactory_url` (exists in .escher/config.json) and version.

File: plugins/monitor/monitor.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+	cmd.Flag("directory", "directory which contains statsd mappings").Required().StringVar(&directory)
+	cmd.Flag("template-url", "URL to s3 which contains the base template for monitoring nomad job").Required().StringVar(&url)
+
+	return cmd
+}
+
+func Run(conf *protocol.Config) error {
+	switch conf.Command {
+	case "monitor":
```
Comment (mohandutt134) [inline plugins/monitor/monitor.go]:
```suggestion
	case "monitorgen":
```

File: plugins/monitor/tmpl/monitoring.nomad.tmpl [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+
+  - job_name: 'simba_metrics'
+    scrape_interval: 300s
+    scrape_timeout: 60s
+    static_configs:
+      - targets: ['simbaMetricsExporter.service.consul:8888']
+  {% endif %}
+
+remote_write:
+  - url: "https://docker-registry.trustingsocial.com/haprom/metrics/?backend=stackdriver"
```
Comment (mohandutt134) [inline plugins/monitor/tmpl/monitoring.nomad.tmpl]: this file has lot of private info like these endpoints, I am closing this PR. Please create another one.

File: .travis.yml [hunk 7]
```diff
@@ -42,3 +42,14 @@ deploy:
     upload-dir: escher/${TRAVIS_OS_NAME}-${TRAVIS_TAG}
     on:
       tags: true
+
+  - provider: s3
```
Comment (mohandutt134) [inline .travis.yml]: remove this as well, again, private info in public repo.

**Comment by mohandutt134** [comment]
@Talina06 ignore private info comments, I have asked Tuyen to make this repo private.

**Comment by meson10** [comment]
@Talina06 @mohandutt134 the plugin is called `monitor`.
`escher monitor` is an appropriately named tool?

**Comment by mohandutt134** [comment]
@meson10 I have added comment above to rename it to `monitorgen`
